src/plugins/telemetryTable/components/table-column-header.vue

```diff
@@ -0,0 +1,182 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+<template>
+<th 
+    :style="{ width: columnWidth + 'px', 'max-width': columnWidth + 'px'}"
+    :draggable="isEditing"
+    @mouseup="sort"
+    v-on="isEditing ? {
+        dragstart: columnMoveStart,
+        drop: columnMoveEnd,
+        dragleave: hideDropTarget,
+        dragover: dragOverColumn
+    } : {}">
+        <div class="c-telemetry-table__headers__content" :class="[
+            isSortable ? 'is-sortable' : '', 
+            isSortable && sortOptions.key === headerKey ? 'is-sorting' : '', 
+            isSortable && sortOptions.direction].join(' ')">
+            <div v-if="isEditing" class="c-telemetry-table__resize-hotzone c-telemetry-table__resize-hotzone--right"
+                @mousedown="startResizeColumn"
+            ></div>
+            <slot></slot>
+        </div>
+</th>
+</template>
+<style lang="scss">
+    @import "~styles/sass-base";
+    @import "~styles/table";
+
+    $hotzone-size: 6px;
+
+    .c-telemetry-table__headers__content {
+        width: 100%;
+    }
+
+    .c-table.c-telemetry-table {
+        .c-telemetry-table__resize-hotzone {
+            display: block;
+            position: absolute;
+            height: 100%;
+            padding: 0;
+            margin: 0;
+            width: $hotzone-size;
+            min-width: $hotzone-size;
+            cursor: col-resize;
+            border: none;
+            right: 0px;
+            margin-right: -$tabularTdPadLR - 1 - $hotzone-size / 2;
+        }
+    }
+</style>
+<script>
+import _ from 'lodash';
+const MOVE_COLUMN_DT_TYPE = 'movecolumnfromindex';
+
+export default {
+    inject: ['openmct'],
+    data() {
+        return {
+            isEditing: this.openmct.editor.isEditing()
+        }
+    },
+    props: {
+        headerKey: String,
+        headerIndex: Number,
+        isHeaderTitle: Boolean,
+        sortOptions: Object,
+        columnWidth: Object,
+        hotzone: Boolean
+    },
+    computed: {
+        isSortable() {
+            return this.sortOptions !== undefined;
+        }
+    },
+    methods: {
+        startResizeColumn(mouseMoveEvent) {
+            this.resizeStartX = mouseMoveEvent.clientX;
+            this.resizeStartWidth = this.columnWidth;
+
+            document.addEventListener('mouseup', (mouseUpEvent)=>{
```

Comment:
What about moving this anonymous callback to a method on the component?  Then you can just get rid of "typed" event names.

Reply:
Good idea. Done.

Reply:
Actually wait.